vowpalwabbit/array_parameters_dense.h

```diff
@@ -49,7 +49,7 @@ class dense_parameters
 public:
   typedef dense_iterator<weight> iterator;
   typedef dense_iterator<const weight> const_iterator;
-  dense_parameters(size_t length, uint32_t stride_shift = 0)
+  dense_parameters(uint64_t length, uint32_t stride_shift = 0)
```

Comment:
>uint64_t [](start = 19, length = 8)

Should this be size_t since its used as an index in array ops.   Similar to how offset is used in most of the codebase?

Reply:
the problem is that this triggers a warning on x86. The weights are always uint64_t.

Reply:
the problem is that this triggers a warning on x86. The weights are always uint64_t.

---
In reply to: [188074118](https://github.com/JohnLangford/vowpal_wabbit/pull/1481#discussion_r188074118) [](ancestors = 188074118)